pingpong/pingpong.go

```diff
@@ -127,26 +127,21 @@ func (pp *pingPonger) pong(ctx context.Context, to protocol.PeerAddress) error {
 	}
 }
 
-func (pp *pingPonger) propagatePing(ctx context.Context, body protocol.MessageBody) error {
-	// TODO: Wrap errors in custom error types.
-
+func (pp *pingPonger) propagatePing(ctx context.Context, sender protocol.PeerID, body protocol.MessageBody) error {
 	peerAddrs, err := pp.dht.PeerAddresses()
 	if err != nil {
 		return err
 	}
```

Comment:
It's almost definitely a bad idea to forward pings to _every_ known peer address (just like doing this for broadcasting isn't a great idea). It would be better to have the `dht` expose something like `dht.RandomPeerAddresses(pp.options.alpha)` that allows us to reduce the network load caused by ping/ponging.

Reply:
You could possibly refactor the `broadcaster` so we can use the same gossiping methodology here, but that feels like overkill right now.